server/game/cards/03_TWI/upgrades/HoldoutBlaster.ts

```diff
@@ -0,0 +1,31 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { Card } from '../../../core/card/Card';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+import { Trait, WildcardCardType, ZoneName } from '../../../core/Constants';
+
+export default class HoldoutBlaster extends UpgradeCard {
+    protected override getImplementationId () {
+        return {
+            id: '7280804443',
+            internalName: 'holdout-blaster',
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.setAttachCondition((card: Card) => !card.hasSomeTrait(Trait.Vehicle));
+
+        this.addWhenPlayedAbility({
+            title: 'You may have attached unit deal 1 damage to a ground unit.',
+            immediateEffect: AbilityHelper.immediateEffects.selectCard({
+                cardTypeFilter: WildcardCardType.Unit,
+                zoneFilter: ZoneName.GroundArena,
+                innerSystem: AbilityHelper.immediateEffects.damage({
+                    optional: true,
+                    amount: 1
```

Comment:
Since the _unit_ is technically dealing the damage, not the upgrade, we need to add `source: context.source.parentCard` here. This matters for cards like the Jango leader